lineproto/reader.go

```diff
@@ -180,7 +180,21 @@ func (r *Reader) ReadLine() ([]byte, error) {
 	}
 }
 
-// ActivateZlib activates zlib deflating.
+// Read reads a byte slice, inflates it if zlib is active, and puts the
+// result into buf.
+func (r *Reader) Read(buf []byte) (int, error) {
+	n, err := r.cur.Read(buf)
+	// n == 0 is needed since Read() can return n > 0 and io.EOF
+	if n == 0 && err == io.EOF && r.zlibOn {
```

Comment:
Looks good, but consider this case as well:
```
zlibOn = true
n = 10
err = EOF
```
Currently, the code will return `10, EOF` to the caller without trying to switch off the zlib.

I would propose to silence the first EOF error, and then let the next call hit the condition you defined.

Reply:
that's true, check if it's better now.